### ISSUE TITLE
Determine if a cluster is TCE using Capability Discovery

### ISSUE THREAD

**Issue by rajathagasthya** [vmware-tanzu/tanzu-framework#383]
**Describe the feature request**

Currently, TKG capability discovery package (`pkg/v1/sdk/capabilities/discovery/tkg`) provides methods to determine if a cluster is TKGm or TKGs. There needs to be a method to also determine if a cluster is a TCE cluster.

Technically, TCE is the "edition", so this brings the question of how edition information is propagated/stored in the cluster given this package primarily relies on Kubernetes API to determine "capabilities".

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

- [x] APIs
- [ ] Addons
- [ ] CLI
- [ ] Docs
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**

**Comment by joshrosso** [comment]
> Technically, TCE is the "edition", so this brings the question of how edition information is propagated/stored in the cluster given this package primarily relies on Kubernetes API to determine "capabilities".

Starting in TCE `v0.11.0`, the edition is exposed in the CAPI `Cluster` object. This object is only available in the `management` cluster.

Reference:

* https://github.com/vmware-tanzu/tanzu-framework/pull/1599

There is an ask, and plan to implement `edition` details in the `tkg-metadata` configmap, starting in TCE v0.12.0 (~May). This would be surfaced in both management and workload clusters.

Reference:

* https://github.com/vmware-tanzu/community-edition/issues/3593

Does the above satisfy this need for our discovery package?

**Comment by rajathagasthya** [comment]
Yes, `edition` details in `tkg-metadata` ConfigMap would be perfect.
